lib/VCMIDirs.cpp

```diff
@@ -164,178 +97,111 @@ class VCMIDirsWIN32 final : public IVCMIDirs
+{
+	if (dirsConfig.has_value())
+		return;
 
-		if (!bfs::is_directory(to))
-		{
-			// IVCMIDirs::init() should create all destination directories.
-			// TODO: Log fact, that we shouldn't be here.
-			bfs::create_directories(to);
-		}
+	wchar_t currentPath[MAX_PATH];
+	GetModuleFileNameW(nullptr, currentPath, MAX_PATH);
 
-		// Why the hell path strings should be end with double null :/
-		auto makeDoubleNulled = [](const bfs::path& path) -> std::unique_ptr<wchar_t[]>
-		{
-			const std::wstring& pathStr = path.native();
-			std::unique_ptr<wchar_t[]> result(new wchar_t[pathStr.length() + 2]);
-
-			size_t i = 0;
-			for (const wchar_t ch : pathStr)
-				result[i++] = ch;
-			result[i++] = L'\0';
-			result[i++] = L'\0';
-
-			return result;
-		};
-
-		auto fromDNulled = makeDoubleNulled(from / L"*.*");
-		auto toDNulled = makeDoubleNulled(to);
-
-		SHFILEOPSTRUCTW fileOp;
-		fileOp.hwnd = GetConsoleWindow();
-		fileOp.wFunc = FO_MOVE;
-		fileOp.pFrom = fromDNulled.get();
-		fileOp.pTo = toDNulled.get();
-		fileOp.fFlags = 0;
-		fileOp.hNameMappings = nullptr;
-		fileOp.lpszProgressTitle = nullptr;
-
-		const int errorCode = SHFileOperationW(&fileOp);
-		if (errorCode != 0) // TODO: Log error. User should try to move files.
-			return false;
-		else if (fileOp.fAnyOperationsAborted) // TODO: Log warn. User aborted operation. User should move files.
-			return false;
-		else if (!bfs::is_empty(from)) // TODO: Log warn. Some files not moved. User should try to move files.
-			return false;
-
-		if (bfs::current_path() == from)
-			bfs::current_path(to);
-
-		// TODO: Log fact that we moved files successfully.
-		bfs::remove(from);
-		return true;
-	};
+	auto configPath = bfs::path(currentPath).parent_path() / "config" / "dirs.json";
 
-	// Retrieves the fully qualified path for the file that contains the specified module.
-	// The module must have been loaded by the current process.
-	// If this parameter is nullptr, retrieves the path of the executable file of the current process.
-	auto getModulePath = [](HMODULE hModule) -> bfs::path
+	if (!bfs::exists(configPath))
 	{
-		wchar_t exePathW[MAX_PATH];
-		DWORD nSize = GetModuleFileNameW(hModule, exePathW, MAX_PATH);
-		DWORD error = GetLastError();
-		// WARN: Windows XP don't set ERROR_INSUFFICIENT_BUFFER error.
-		if (nSize != 0 && error != ERROR_INSUFFICIENT_BUFFER)
-			return bfs::path(std::wstring(exePathW, nSize));
-		// TODO: Error handling
-		return bfs::path();
-	};
+		dirsConfig = std::nullopt;
+		return;
+	}
 
-	// Moves one directory contents to another directory
-	// Shows user the "moving file dialog" and ask to resolve conflicts.
-	// It takes into account that 'from' path can contain current executable.
-	// If necessary closes program and starts update script.
-	auto advancedMoveDirIfExists = [getModulePath, moveDirIfExists](const bfs::path& from, const bfs::path& to) -> bool
+	std::ifstream in(configPath.string(), std::ios::binary);
```

Comment:
Haven't investigated the reason, but `string()` fails to open file with UTF-8 characters in path on Windows 
```suggestion
	std::ifstream in(configPath.c_str(), std::ios::binary);
```